django/contrib/postgres/search.py

```diff
@@ -219,6 +221,53 @@ def as_sql(self, compiler, connection, function=None, template=None):
         return sql, extra_params + params
 
 
+class SearchHeadline(Func):
+    function = 'ts_headline'
+    output_field = TextField()
+    options = None
+
+    def __init__(self, expression, query, **extra):
+        if not hasattr(query, 'resolve_expression'):
+            query = SearchQuery(query)
+        self.config = query.config
+        self.options = extra.get('options', self.options)
+        super().__init__(expression, query, **extra)
```

Comment:
Can we validate at least names of options?

Reply:
Sure, in my first draft I actually remapped all the CamelCased postgres option names to more pythonic names:

```python
option_mapping = {
    'start_sel': 'StartSel',
    'stop_sel': 'StopSel',
    'max_words': 'MaxWords',
    'min_words': 'MinWords',
    'short_word': 'ShortWord',
    'highlight_all': 'HighlightAll',
    'max_fragments': 'MaxFragments',
    'fragment_delimiter': 'FragmentDelimiter',
}
```

But then it felt like the need to more of less duplicate the documentation from the postgres docs for our version of the options would end up being a little bit too much. 

Do you think we should use the postgres names and validate them or keep our own mapping?

Reply:
IMO we should check options with PostreSQL names.